records_mover/session.py

```diff
@@ -185,11 +199,10 @@ def __init__(self,
                                  default_db_facts=default_db_facts,
                                  default_boto3_session=default_boto3_session,
                                  default_gcp_creds=default_gcp_creds,
-                                 default_gcs_client=default_gcs_client)
-
-        if scratch_s3_url is PleaseInfer.token:
-            scratch_s3_url = _infer_scratch_s3_url(session_type)
+                                 default_gcs_client=default_gcs_client,
+                                 scratch_s3_url=scratch_s3_url)
 
+        # TODO: get this from creds
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker